client/utils/guidance.py

```diff
@@ -59,23 +58,27 @@
 }
 
 class Guidance:
-    def __init__(self, current_working_dir):
+    def __init__(self, current_working_dir,assignment = None):
         """
         Initializing everything we need to the default values. If we catch
         an error when opening the JSON file, we flagged it as error.
         """
         self.tg_id = -1
+        if assignment:
+            self.assignment = assignment.name.replace(" ","")
+        else:
+            self.assignment = ""
         self.misU_count_dict = {} #Maps the type of misunderstanding seen to the number of times seen
         self.current_working_dir = current_working_dir
         try:
             with open(current_working_dir + OK_GUIDANCE_FILE,"r") as f:
                 self.guidance_json = json.load(f)
             self.load_error = False
         except IOError as e:
-            log.warning("Failed to read .ok_guidance file in guidance.py. IOError")
+            log.error("Failed to read .ok_guidance file in guidance.py. IOError")
             self.load_error = True
         except ValueError as e:
-            log.warning("Failed to read .ok_guidance file in guidance.py. ValueError")
+            log.error("Failed to read .ok_guidance file in guidance.py. ValueError")
```

Comment:
If it's relatively straightforward to get the text message out of the `e` please use that text instead of just "ValueError".

Comment also applies to other places with errors.